File: c++/456. 132 Pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        int min_val = INT_MIN, n = nums.size();
        stack<int> st;
        for (int i = n-1; i >= 0; i--) {
            if (nums[i] < min_val) return true;
            while (!st.empty() && st.top() < nums[i]) {
                min_val = st.top();
                st.pop();
            }
            st.push(nums[i]);
        }
        return false;
    }
};

File: go/src/leetcode/456. 132 Pattern/456. 132 Pattern.go
package main

import "math"

func find132pattern(nums []int) bool {
	minVal := math.MinInt32
	st := []int{}
	for i := len(nums)-1; i >= 0; i-- {
		if minVal > nums[i] {
			return true
		}
		for len(st)>0 && st[len(st)-1]<nums[i] {
			minVal = st[len(st)-1]
			st = st[:len(st)-1]
		}
		st = append(st, nums[i])
	}
	return false
}
